agent/engine/task_manager.go

```diff
@@ -763,10 +763,20 @@ func (mtask *managedTask) progressTask() {
 	mtask.waitForTransition(transitions, transitionChange, transitionChangeEntity)
 	// update the task status
 	changed := mtask.UpdateStatus()
+
+	// If knownStatus changed, let it be known
+	mtask.emitEventIfStatusChanged(changed)
```

Comment:
How about something like:
```
if changed {
 mtask.emitStatusChangeEvent()
}

func (mtask *managedTask) emitStatusChangeEvent() {
..
}
```

Reply:
there's already method for that exactly. i put the branching logic behind a method so we can test it.

Reply:
The new branching logic that was added was the check for terminal status and what kind of reason was emitted, isn't it? Unit testing that should be fine I think.